tfjs-backend-webgpu/src/kernels/gather_webgpu.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {getCoordsDataType} from '../shader_preprocessor';
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
+
+import {WebGPUProgram} from './webgpu_program';
+
+export class GatherProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames: string[] = ['A', 'indices'];
+  workPerThread = 4;
+  workGroupSize: [number, number, number] = [64, 1, 1];
+  rank: number;
+  aShape: number[];
+
+  constructor(aShape: number[], outputShape: number[]) {
+    this.outputShape = aShape.slice();
+    this.aShape = aShape;
+    this.outputShape = outputShape;
+    this.rank = outputShape.length;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize,
+        [this.workPerThread, 1, 1]);
```

Comment:
`this.workPerThread` is not used in your shader. But you define it as a non-one value.